tests/instruments/test_instrument.py

```diff
@@ -234,18 +258,89 @@ class Fake(FakeInstrument):
 
 
 @pytest.mark.parametrize(
-    'set_command, given, expected',
-    [("%d", 5, 5),
-     ("%d, %d", (5, 6), [5, 6]),  # input has to be a tuple, not a list
+    'set_command, given, expected, dynamic',
+    [("%d", 5, 5, False),
+     ("%d", 5, 5, True),
+     ("%d, %d", (5, 6), [5, 6], False),  # input has to be a tuple, not a list
+     ("%d, %d", (5, 6), [5, 6], True),  # input has to be a tuple, not a list
      ])
-def test_fakeinstrument_control(set_command, given, expected):
+def test_fakeinstrument_control(set_command, given, expected, dynamic):
     """FakeInstrument's custom simple control needs to process values correctly.
     """
     class Fake(FakeInstrument):
         x = FakeInstrument.control(
             "", set_command, "",
+            dynamic=dynamic,
         )
 
     fake = Fake()
     fake.x = given
     assert fake.x == expected
+
+
+def test_instrument_dynamic_parameter():
+    class GenericInstrument(FakeInstrument):
+        fake_ctrl = Instrument.control(
+            ":PARAM?;", ":PARAM %e Hz;",
+            """ A property that represents ...
+            """,
+            validator=strict_range,
+            values=(1, 10),
+            dynamic=True
+        )
+        fake_setting = Instrument.setting(
+            ":PARAM1 %e Hz;",
+            """ A property that represents ...
+            """,
+            validator=strict_range,
+            values=(1, 10),
+            dynamic=True
+        )
+        fake_measurement = Instrument.measurement(
+            "",
+            """ A property that represents ...
+            """,
+            values={'X': 1, 'Y': 2, 'Z': 3},
+            map_values=True,
+            dynamic=True
+        )
+
+    class SpecificInstrument1(GenericInstrument):
+        # Set properties' values specific to SpecificInstrument1
+        fake_ctrl_values = (1, 10)
+        fake_setting_values = (1, 10)
+        fake_measurement_values = {'X': 1, 'Y': 2, 'Z': 3}
```

Comment:
These are identical to GenericInstrument -- is this needed?

Reply:
I'll rework this a bit and propose a changed implementation

Reply:
Ok, I think your proposed test implementation clear this, even though I am not sure why I wrote the test like this :)